R/contribution_interpretation.R

```diff
@@ -0,0 +1,324 @@
+    cg_subset <- medecom.set@parameters$cg_subsets[1]
+  }else if (!(cg_subset %in%  medecom.set@parameters$cg_subsets)){
+    stop("Specified value for cg_subset not in medecom.set")
+  }
+  if(is.null(K)){
+    K <- medecom.set@parameters$Ks[1]
+  }else if (!(K %in%  medecom.set@parameters$Ks)){
+    stop("Specified value for K not in medecom.set")
+  }
+  if(is.null(lambda)){
+    lambda <- medecom.set@parameters$lambdas[1]
+  }else if (!(lambda %in%  medecom.set@parameters$lambdas)){
+    stop("Specified value for lambda not in medecom.set")
+  }
+  ret.list <- list()
+  p.vals <- link.to.traits(medecom.set=medecom.set,cg_subset=cg_subset,K=K,lambda=lambda,rnb.set=rnb.set,test.fun=test.fun)
+  plot <- plot.p.val.heatmap(p.vals)
+  ret.list[["qualitative"]] <- plot
+  cors <- quantitative.trait.association(medecom.set=medecom.set,cg_subset=cg_subset,K=K,lambda=lambda,rnb.set=rnb.set)
+  plot <- plot.correlation.heatmap(cors)
+  ret.list[["quantitative"]] <- plot
+  return(ret.list)
+}
+
+#' link.to.traits
+#' 
+#' This routine performs a statistical test to determine if the difference in LMC contributions is different for the sample groups
+#' defined in \code{rnb.set}.
+#' 
+#' @param medecom.set An object of type \code{\link{MeDeComSet}} as the result of \code{\link{runMeDeCom}} containing LMCs and their
+#'                     proportions in the samples. The Set can contain multiple runs for different values of K and lambda.
+#' @param cg_subset The subset of sites used for computing LMCs in \code{medecom.set}.
+#' @param K The K parameter, determining the number of LMCs to extract from \code{medecom.set}.
+#' @param lambda The lambda parameter, determining the regularization used for the LMCs in \code{medecom.set}.
+#' @param rnb.set An object of type \code{\link{RnBSet}} containing methylation data and metadata for the same samples for which 
+#'                 \code{medecom.set} was computed or a data.frame of sample annotations (ann_S)
+#' @param test.fun Test statistic used to compute p-values of differences between LMC contributions in pairwise sample comparisons.
+#'                  Defaults to \code{t.test}.
+#'                  
+#' @return A list with an element for each sample grouping defined by \code{\link{rnb.sample.groups}}.
+#' 
+#' @details Each element in the returned list is of length \code{K}, displaying the p-value of the statistical assocation of the
+#'           contributions of the corresponding LMC to the sample grouping. The p-values are produced by comparing the LMC 
+#'           contributions in all sample comparisons defined by \code{\link{rnb.sample.groups}} on \code{rnb.set}. The employed test
+#'           statistic for pariwise comparison can be specified by \code{test.fun}, for groups defining more than one group
+#'           \code{\link{kruskal.test}} is employed.
+#' @author Michael Scherer
+#' 
+#' @noRd
+
+link.to.traits <- function(medecom.set,cg_subset,K,lambda,rnb.set,test.fun=t.test){
```

Comment:
Should we create more generic functions for this, which use phenotype data frames as input? RnBeads wrappers can be added on top?